core/src/main/java/io/github/tcdl/Collector.java

```diff
@@ -135,7 +136,12 @@ protected void end() {
         cancelAckTimeoutTask();
         cancelResponseTimeoutTask();
 
-        collectorManager.unsubscribe(this);
+        Consumer consumer = channelManager.findOrCreateConsumer(topic);
+        consumer.removeSubscriber(requestMessage.getCorrelationId());
+        if(consumer.getMessageHandlersByCorrelationId().isEmpty()) {
+            channelManager.unsubscribe(topic);
+        }
```

Comment:
why do we need to know about getMessageHandlersByCorrelationId in Collector? this call is not obvious.